### DISCUSSION TITLE
Make better use of tags in pyperformance

### DISCUSSION THREAD

**Discussion by mdboom** [faster-cpython/ideas#395] [Ideas]
It's becoming obvious that:

- The pyperformance suite needs more benchmarks that are more similar to real-world workloads, and we should lean into optimizing for these and using these to report progress.
- Microbenchmarks of a particular feature are also useful and belong in the benchmark suite, but we shouldn't over-optimize for them or use them as a (misleading) indicator of overall progress.

It seems that one way to address this would be to lean into "tags" more in the pyperformance/pyperf ecosystem. pyperformance already allows for tags in each benchmark's `pyproject.yaml`. 

I propose we:

1) Output the tags for each benchmark in the benchmark results in the `metadata` dictionary.
2) `pyperf` `compare_to` would then calculate the geometric mean for each subset of benchmarks for each tag found in the results, as well as "all" benchmarks (existing behavior). This could be behind a flag if backward compatibility matters.

# Alternatives:

We could instead use the nested benchmark heirarchy, rather than tags. Personally, I think tags is easier to understand and more flexible (a benchmark could be associated with multiple tags).

**Comment by ericsnowcurrently** [comment]
Could we move this issue to the pyperformance repo?

**Reply by mdboom** [reply]
Sure --> https://github.com/python/pyperformance/issues/208

**Reply by ericsnowcurrently** [reply]
Thanks!
